packages/ui-utils/src/call-chat-api.ts

```diff
@@ -106,3 +106,91 @@ export async function callChatApi({
     }
   }
 }
+
+export async function resumeChatApi({
+  api,
+  body,
+  fetch = getOriginalFetch(),
+  onResponse,
+  restoreMessagesOnFailure,
+  streamProtocol = 'data',
+  onUpdate,
+  onFinish,
+  onToolCall,
+  generateId,
+  lastMessage,
+}: {
+  api: string;
+  body: Record<string, any>;
+  fetch: ReturnType<typeof getOriginalFetch> | undefined;
+  onResponse: ((response: Response) => void | Promise<void>) | undefined;
+  restoreMessagesOnFailure: () => void;
+  streamProtocol: 'data' | 'text' | undefined;
+  onUpdate: (options: {
+    message: UIMessage;
+    data: JSONValue[] | undefined;
+    replaceLastMessage: boolean;
+  }) => void;
+  onFinish: UseChatOptions['onFinish'];
+  onToolCall: UseChatOptions['onToolCall'];
+  generateId: IdGenerator;
+  lastMessage: UIMessage | undefined;
+}) {
+  const { id } = body;
+
+  const response = await fetch(`${api}?chatId=${id}`, {
+    method: 'GET',
+  });
+
+  if (onResponse) {
+    try {
+      await onResponse(response);
+    } catch (err) {
+      throw err;
+    }
```

Comment:
why is try catch rethrow needed? isn't `await onResponse(response);` sufficient and the same even in the error case?

Reply:
removed this